### PR TITLE
Feature: Translatable app attributes

### PR DESCRIPTION
**Pull request by opyh**
This allows to translate app attributes and main menu links.

It includes a (copied) new version of the i18n-with-transifex code.

To make a collection translatable, you can use the code like this:
```javascript
import getDefaultTranslationStrategy from '../../../../server/i18n-new/getDefaultTranslationStrategy';
import makeCollectionTranslatable from '../../../../server/i18n-new/makeCollectionTranslatable';
import { AppLinks } from '../app-links';

const translatablePropertyNames = [
  'label',
  'url',
];

const collection = AppLinks;
makeCollectionTranslatable({ collection, translationStrategy: getDefaultTranslationStrategy({ collection, translatablePropertyNames })})
```

If you need to change the way how translations are fetched from or saved to the local system, you can implement your own translation strategy.

The refactoring decouples transifex logic from MongoDB logic, making the API more usable.

The `apps.json` API endpoint now includes app links as related objects per default. As the links are not saved as array anymore, each link has an `order` attribute.

I'll deploy this live as soon as I have adapted frontend code for this.

### CODE REVIEW

File: both/api/app-links/app-links.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+export const AppLinks = new Mongo.Collection('AppLinks');
+
+
+AppLinks.schema = new SimpleSchema({
+  appId: { type: SimpleSchema.RegEx.Id },
+  label: localizedStringSchema,
+  url: localizedStringSchema,
+  order: { type: Number },
+});
```
[the remark this answers is not in the document]
Comment (opyh) [reply both/api/app-links/app-links.js]: Yes, probably better than having more properties in the config for this!

File: tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,20 @@
+{
+  "compilerOptions": {
+    "allowJs": true,
+    "lib": ["es6", "es2017"],
+    "module": "commonjs",
+    "target": "es5",
+    "moduleResolution": "node",
+    "experimentalDecorators": true,
+    "emitDecoratorMetadata": true,
+    "sourceMap": true
```
[the remark this answers is not in the document]
Comment (mutaphysis) [reply tsconfig.json]: Just noticed - there is a `strict` flag for all this.

File: server/i18n-new/returnNullIf404.ts [hunk 3]
```diff
@@ -0,0 +1,13 @@
+export default function returnNullIf404(fn: Function) {
```
[the remark this answers is not in the document]
Comment (opyh) [reply server/i18n-new/returnNullIf404.ts]: (cough)

File: tsconfig.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (opyh) [reply tsconfig.json]: Works :)

**Pull request merged**